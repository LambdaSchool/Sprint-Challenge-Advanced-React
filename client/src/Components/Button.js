import React from 'react';
import useDarkMode from '../Hook/useDarkMode';

const Button = ({label}) => {
    const [darkMode, setDarkMode] = useDarkMode(false);
    const toggleMode = e => {
      e.preventDefault();
      setDarkMode(!darkMode);
      };
    return (
     <div>
        <h3> {label} </h3>
         <button className= 'btn'

            onClick={toggleMode}
            className={darkMode ? 'toggle toggled' : 'toggle'}>
         </button>
     </div>
    )};

export default Button;