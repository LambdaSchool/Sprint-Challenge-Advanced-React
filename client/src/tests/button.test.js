import React from 'react';
import { render, cleanup,toHaveTextContent} from "@testing-library/react";
import Button from '../Components/Button';
import renderer from 'react-test-renderer';


afterEach(cleanup)
it ('renders button correctly',() => {
    const{getByTestId} = render(<Button label='Dark Mode'/>);
    expect (getByTestId("button"))})


// it ('renders button correctly',() => {
//     const{getByTestId} = render(<Button label='save'/>);
//     expect (getByTestId("button")).toHaveTextContent('save');
// })


it('matches snapshot', ()=> {
   const tree = renderer.create(<Button label = 'save'></Button>).toJSON();
   expect(tree).toMatchSnapshot();
})