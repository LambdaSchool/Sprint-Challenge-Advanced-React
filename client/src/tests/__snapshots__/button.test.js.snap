// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h3>
     
    save
     
  </h3>
  <button
    className="toggle"
    data-testid="button"
    onClick={[Function]}
  />
</div>
`;
